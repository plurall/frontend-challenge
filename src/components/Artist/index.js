export { default as ArtistDetailedCard } from './ArtistDetailedCard'
export { default as AlbumCard } from './AlbumCard'
export { default as AlbumsList } from './AlbumsList'
export { default as LoadingArtistDetailedCard } from './LoadingArtistDetailedCard'
export { default as LoadingAlbumsList } from './LoadingAlbumsList'
export { default as LoadingAlbumCard } from './LoadingAlbumCard'
