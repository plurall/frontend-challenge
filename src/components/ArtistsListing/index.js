export { default as NoArtistsMessage } from './NoArtistsMessage'
export { default as ArtistSmallCard } from './ArtistSmallCard'
export { default as ArtistsList } from './ArtistsList'
export { default as SearchRemainingLetters } from './SearchRemainingLetters'
export { default as LoadingArtistsList } from './LoadingArtistsList'
export { default as LoadingArtistSmallCard } from './LoadingArtistSmallCard'
