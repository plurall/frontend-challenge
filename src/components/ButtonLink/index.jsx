import React from 'react';
import PropTypes from 'prop-types'
import magnifierGlassSVG from "../../assets/icons/magnifier-glass.svg"

import {
  button_link as buttonLink,
  details_image as detailsImage
 } from './ButtonLink.module.css';
import { Link } from 'react-router-dom';

const ButtonLink = ({
  id,
  name,
  linkTo,
  image = {
    url: magnifierGlassSVG,
    alt: "image default"
  }
}) => {

  return (
    <Link
      id={id}
      name={name}
      className={buttonLink}
      to={linkTo}
    >
      <img
        className={detailsImage}
        src={image.url}
        alt={image.alt}
      />
    </Link>
  );
}

ButtonLink.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  linkTo: PropTypes.string.isRequired,
  image: PropTypes.shape({
    url: PropTypes.string,
    alt: PropTypes.string
  })
}

export default ButtonLink;
