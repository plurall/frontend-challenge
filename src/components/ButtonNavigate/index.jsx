import React from 'react';
import PropTypes from 'prop-types'

import {
  button
} from './ButtonNavigate.module.css';

const ButtonNavigate = ({
  id,
  name,
  disabled = false,
  onClick,
  image
}) => {
  return (
    <button
      id={id}
      name={name}
      disabled={disabled}
      onClick={onClick}
      className={button}
    >
      <img src={image.url} alt={image.alt} />
    </button>
  );
}

ButtonNavigate.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  image: PropTypes.shape({
    url: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired
  })
}

export default ButtonNavigate;
