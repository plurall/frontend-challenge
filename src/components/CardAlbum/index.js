export { default as CardAlbum } from './CardAlbum'
