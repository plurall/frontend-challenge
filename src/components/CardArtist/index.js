export { default as CardArtist } from './CardArtist'
