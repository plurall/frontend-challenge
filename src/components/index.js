export { default as Layout } from './Layout'
export { default as SubHeader } from './SubHeader'
export { default as PrivateRoute } from './PrivateRoute'
export { default as Loader } from './Loader'
export { default as Error } from './Error'
