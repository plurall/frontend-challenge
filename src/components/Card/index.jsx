
/* eslint-disable */
import React from 'react'
import PropTypes from 'prop-types'
import { BaseRoutes } from 'routes/BaseRoutes'
import magnifierGlassSVG from "../../assets/icons/magnifier-glass.svg"
import userDefault from "../../assets/icons/user-default.svg"
import ButtonLink from '../../components/ButtonLink'
import {
  image,
  folder,
  wrapper,
  banner,
  footer,
  footer_artist as footerArtist,
  footer_folder as footerFolder,
  wrapper_artist as wrapperArtist,
  wrapper_folder as wrapperFolder,
} from "./Card.module.css"

const Card = ({
  id,
  photo,
  name,
  description = "Artista",
  type = "artist"
}) => {
  const isArtist = type === "artist"
  return (
    <div className={`${wrapper} ${description === "Artista" ? wrapperArtist : wrapperFolder}`}>

      {isArtist ?
        <ButtonLink
          id={`link_${id}`}
          name={name}
          linkTo={BaseRoutes.artist.replace(":id", id)}
          image={{
            url: magnifierGlassSVG,
            alt:"lupa"
          }}
        />
      : ""}

      <img
        className={`${banner} ${ isArtist ? image : folder}`}
        src={photo || userDefault} alt="foto do artista"
      />

      <footer className={`${footer} ${ isArtist ? footerArtist : footerFolder}`}>
        <strong>{name}</strong>
        <span >{description}</span>
      </footer>
    </div>
  )
}

Card.propTypes = {
  id: PropTypes.string.isRequired,
  photo: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string,
  type: PropTypes.string,
}

export default Card
