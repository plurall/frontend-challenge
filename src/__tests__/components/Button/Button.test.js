import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from "react-router-dom"; // our router

import { Button } from 'components';

describe('Renderização do Button', () => {

 it('deve ser definido o componente', () => {

   expect(Button).toBeDefined();
 });

 it('deve equivaler ao snapshot', () => {
  const tree = shallow(
    <Button
      href='/busca'
      text='button test'
    />
  );

  expect(tree).toMatchSnapshot();
});

 it('deve renderizar corretamente', () => {
   const tree = shallow(
      <Button
        href='/busca'
        text='button test'
      />
   );

   expect(tree).toMatchSnapshot();
 });

  it("deve ter o link passado como propriedade", () => {
    const link = '/busca';
    const element = shallow(<Button text={'Buscar'} href={link} />);

    expect(element.prop('to')).toEqual(link);
  });

  it("deve ter o link passado como propriedade", () => {
    const link = '/busca';

     const wrapper = mount(
      <MemoryRouter>
        <Button text={'Buscar'} href={link} />
      </MemoryRouter>
    );

    expect(wrapper.find('[href="/busca"]').length).toBe(2);
  });
});
