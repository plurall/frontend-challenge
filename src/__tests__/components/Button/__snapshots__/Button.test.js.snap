// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderização do Button deve equivaler ao snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    href="/busca"
    text="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "button test",
      "className": "c-btn",
      "replace": false,
      "to": "/busca",
    },
    "ref": null,
    "rendered": "button test",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "button test",
        "className": "c-btn",
        "replace": false,
        "to": "/busca",
      },
      "ref": null,
      "rendered": "button test",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Renderização do Button deve renderizar corretamente 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    href="/busca"
    text="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "button test",
      "className": "c-btn",
      "replace": false,
      "to": "/busca",
    },
    "ref": null,
    "rendered": "button test",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "button test",
        "className": "c-btn",
        "replace": false,
        "to": "/busca",
      },
      "ref": null,
      "rendered": "button test",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
